### PR TITLE
Migrate to pyproject.toml

### PR DESCRIPTION
**Pull request by jsoucheiron**
Upgrading the project to use pyproject.toml
It upgrades several core components of the Makefile but has no real changes in the library besides dropping python 3.7 support (it's EOL anyway).
Changes:
- Move from black and isort to ruff
- Use uv to freeze dependencies
- Remove setup.py and move configuration to pyproject.toml
- Dynamically obtain version from source control
- Generate 3 independent frozen requirement files: normal, dev, docs.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,8 +1,132 @@
...
+[project.scripts]
+cfripper = "cfripper.cli:cli"
+
+[project.optional-dependencies]
+dev = [
+    "moto[all]>=5",
+    "pytest-cov>=2.5.1",
+    "pytest>=3.6",
+    "ruff",
+    "uv",
```
Comment (ignaciobolonio) [inline pyproject.toml:60]: no pinned dependencies?

**Review by ignaciobolonio — COMMENTED** [review]
Just left a comment, otherwise, looking good

**Review by ignaciobolonio — APPROVED** [review]
No need to pin dev dependencies

File: pyproject.toml [hunk 2]
```diff
@@ -1,8 +1,132 @@
...
+source = ["cfripper"]
+
+[tool.setuptools]
+include-package-data = false
+
+[tool.setuptools.packages.find]
+# needed only because we did not adopt src layout yet
+include = ["cfripper*"]
+
+[tool.setuptools_scm]
```
Comment (w0rmr1d3r) [inline pyproject.toml:132]: Question only: Is this needed if this section is empty?

**Review by w0rmr1d3r — APPROVED** [review]

File: pyproject.toml (hunk 2 above)
Comment (jsoucheiron) [reply pyproject.toml:132]: According to the documentation examples of the tool seems to be needed.

**Comment by jsoucheiron** [comment]
I'll let you merge it at your earliest convenience, as I don't have time to verify everything properly and there's a non-negligible chance that this will break

**Pull request merged**